src/Microsoft.AspNet.Hosting/Internal/WebApplication.cs

```diff
@@ -98,14 +127,14 @@ public virtual IApplication Start()
             _applicationLifetime.NotifyStarted();
             logger.Started();
 
-            return new Application(ApplicationServices, Server.Features, new Disposable(() =>
-            {
-                logger.Shutdown();
-                _applicationLifetime.StopApplication();
-                Server.Dispose();
-                _applicationLifetime.NotifyStopped();
-                (_applicationServices as IDisposable)?.Dispose();
-            }));
+            return new Disposable(() =>
```

Comment:
It was `IDisposable Start()` from a long time ago, but the pattern changed. The builder was changed to return a type that implemented IDisposable, and also had a public void Start method, but that object which was returned no longer supports multiple calls to Start so there's no reason for each call to Start to return a new disposable.